flink-core/src/main/java/org/apache/flink/types/Row.java

```diff
@@ -66,10 +66,11 @@ public int getArity() {
 	 * Gets the field at the specified position.
 	 * @param pos The position of the field, 0-based.
 	 * @return The field at the specified position.
-	 * @throws IndexOutOfBoundsException Thrown, if the position is negative, or equal to, or larger than the number of fields.
+	 * Return null if the position is equal to, or larger than the number of fields.
+	 * @throws IndexOutOfBoundsException Thrown, if the position is negative.
 	 */
 	public Object getField(int pos) {
-		return fields[pos];
+		return pos >= fields.length ? null : fields[pos];
```

Comment:
This will cause overhead for basically every operation and should not be done to support a minor feature.
If you expect that you might receive a `Row` which violates the expected schema and you want to avoid an `IndexOutOfBoundsException` you should rather check `getArity()`.